gandalf.js

```diff
@@ -1,35 +1,39 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.